py/builtinimport.c

```diff
@@ -305,8 +334,8 @@ mp_obj_t mp_builtin___import__(size_t n_args, const mp_obj_t *args) {
         const char *p = this_name + this_name_l;
         if (!is_pkg) {
             // We have module, but relative imports are anchored at package, so
-            // go there.
-            chop_component(this_name, &p);
+            // take off another level.
+            level++;
```

Comment:
This is a neat (albeit minor) code-space optimisation, which could be taken independent of this PR.